### ISSUE TITLE
Feature Request:  Deprecation Function that automates Deprecation Strategy

### ISSUE THREAD

**Issue by bms63** [pharmaverse/admiraldev#36]
### Feature Idea

A function that helps to remind us when a function should be removed from the code-base.

Essentially a wrapper function around deprecate_stop, deprecate_warn but also causes the package to be unable to build once the function is no longer kept inside admiral.

**Comment by bms63** [comment]
While our manual process is a bit cumbersome, perhaps automating will prove even more cumbersome and challenging and a waste of effort.

I haven't heard of anyone else doing this in other tidyverse or R packages?  @pharmaverse/admiral has anyone seen this?  If not, I think we should close this one out.

**Comment by ddsjoberg** [comment]
I agree it's best here to stick with the standard manual process. While perhaps cumbersome now, we'll have fewer and fewer deprecations in our future. 🤞🏼

**Comment by bms63** [comment]
Going to close - but if folks come up with examples of seeing this in the wild or think it should remain in the backlog feel free to re-open.
